tests/lib/rules/no-array-constructor.js

```diff
@@ -469,21 +424,74 @@ ruleTester.run("no-array-constructor", rule, {
 			},
 		].map(props => ({
 			...props,
+			output: props.code.replace(
+				/(new )?Array\((?<args>.*?)\)/su,
+				"[$<args>]",
+			),
 			errors: [
 				{
-					messageId: "preferLiteral",
-					suggestions: [
-						{
-							messageId: "useLiteral",
-							output: props.code.replace(
-								/(new )?Array\((?<args>.*?)\)/su,
-								"[$<args>]",
-							),
-						},
-					],
+					messageId: "useLiteral",
 				},
 			],
 		})),
+		{
+			code: "new /*a*/ /*b*/ Array /*c*/ /*d*/ () /*e*/ /*f*/ ; /*g*/ /*h*/",
+			output: " /*a*/ /*b*/ [] /*c*/ /*d*/  /*e*/ /*f*/ ; /*g*/ /*h*/",
```

Comment:
Shouldn't `/*c*/ /*d*/` be before `[]` in the output, since they were before `()` (in general case, before array elements) in the original code?

Reply:
Also, I think it's fairly unusual to have comments between `new` and `Array`, or between `Array` and `(`, and it's not clear what these comments relate to (maybe to the Array constructor itself rather than the array), so this might be another case when it seems safer to keep the fix as suggestion instead of autofix.